src/main/java/org/terasology/core/world/generator/trees/TreeGeneratorCactus.java

```diff
@@ -22,15 +22,16 @@
 
 /**
  * Cactus generator.
- *
  */
 public class TreeGeneratorCactus extends AbstractTreeGenerator {
 
+    private static final int MAX_HEIGHT = 4;
+
     private BlockUri cactusType;
 
     @Override
     public void generate(BlockManager blockManager, CoreChunk view, Random rand, int posX, int posY, int posZ) {
-        for (int y = posY; y < posY + 3; y++) {
+        for (int y = posY; y < posY + rand.nextInt(1, MAX_HEIGHT); y++) {
```

Comment:
I believe the height==1 cacti look a bit lost... maybe consider a min height of 2

Reply:
Hm, I especially like them :thinking: they remind me of these small, stubby cacti (https://de.wikipedia.org/wiki/Echinocactus_grusonii)

Reply:
fair, your PR ;)